src/cmd/src/standalone.rs

```diff
@@ -347,47 +344,32 @@ impl StartCommand {
 
         opts.user_provider.clone_from(&self.user_provider);
 
-        let metadata_store = opts.metadata_store.clone();
-        let procedure = opts.procedure.clone();
-        let frontend = opts.clone().frontend_options();
-        let logging = opts.logging.clone();
-        let wal_meta = opts.wal.clone().into();
-        let datanode = opts.datanode_options().clone();
-
-        Ok(Options::Standalone(Box::new(MixOptions {
-            procedure,
-            metadata_store,
-            data_home: datanode.storage.data_home.to_string(),
-            frontend,
-            datanode,
-            logging,
-            wal_meta,
-        })))
+        Ok(Options::Standalone(Box::new(opts)))
     }
 
     #[allow(unreachable_code)]
     #[allow(unused_variables)]
     #[allow(clippy::diverging_sub_expression)]
-    async fn build(self, opts: MixOptions) -> Result<Instance> {
+    async fn build(self, opts: StandaloneOptions) -> Result<Instance> {
         info!("Standalone start command: {:#?}", self);
         info!("Building standalone instance with {opts:#?}");
 
-        let mut fe_opts = opts.frontend;
+        let mut fe_opts = opts.clone().frontend_options();
         #[allow(clippy::unnecessary_mut_passed)]
         let fe_plugins = plugins::setup_frontend_plugins(&mut fe_opts) // mut ref is MUST, DO NOT change it
             .await
             .context(StartFrontendSnafu)?;
 
-        let dn_opts = opts.datanode;
+        let dn_opts = opts.clone().datanode_options();
 
         set_default_timezone(fe_opts.default_timezone.as_deref()).context(InitTimezoneSnafu)?;
 
+        let data_home = dn_opts.storage.data_home.clone();
```

Comment:
Seems we don't need to clone it?

```suggestion
        let data_home = &dn_opts.storage.data_home;
```

Reply:
you're right, let me remove it.